framework/include/actions/MaterialOutputAction.h

```diff
@@ -182,5 +169,9 @@ template <typename T>
 bool
 MaterialOutputAction::hasFunctorProperty(const std::string & property_name)
 {
-  return _problem->hasFunctorWithType<T>(property_name, 0);
+  // functors support a limited set of types
+  if constexpr (std::is_same_v<T, Real> || std::is_same_v<T, RealVectorValue>)
```

Comment:
This change is confusing. Now we're mixing the idea of having a functor property vs. whether the output action supports it. Additionally this limiting conditional isn't quite right. We should also be able to call this with the equivalent AD types

Reply:
Yeah, I hadn't seen your review yet, and had second thoughts about this whole design. The next iteration will have a separate class for the functor outputs.